tests/samplers/test_sampler.py

```diff
@@ -56,7 +57,7 @@ def _do_sample(
             SequenceGroupMetadata(
                 request_id=f"test_{i}",
                 is_prompt=True,
-                seq_data={0: SequenceData([1, 2, 3])},
+                seq_data={0: SequenceData(array("I", [1, 2, 3]))},
```

Comment:
I think we can also accept list input, and do the convert inside `SequenceData` constructor?

I'd like to reduce unnecessary files touched by this PR.

Reply:
The issue is I found msgspec doesn't allow me to have input of Union[List, array]. Technically we can just implicitly convert all list though (I think the type there is just a hint) if you prefer that way